### PR TITLE
DOC-1050: RedisJSON Quick Start Rewrite

### PR DESCRIPTION
**Pull request by cmilesb**
Staging link: https://docs.redis.com/staging/jira-doc-1050/modules/redisjson/redisjson-quickstart/

### CODE REVIEW

File: content/modules/redisjson/redisjson-quickstart.md [hunk 1]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+- [redis-py](https://github.com/redis/redis-py) client library v4.0.0 or greater
 
-## Quick start with redis-cli
+## RedisJSON with redis-cli
 
-These examples use redis-cli as the Redis client to show how easy
-it is. The first RedisJSON command to try out is JSON.SET, which sets a
-Redis key with a JSON value. All JSON values can be used, for example a
-string:
+The `redis-cli` command line tool comes packaged with Redis. You can use it to connect to your Redis database and test the following RedisJSON commands.
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:26]:
```suggestion
The [`redis-cli`](https://redis.io/docs/manual/cli/) command line tool comes packaged with Redis. You can use it to connect to your Redis database and test the following RedisJSON commands.
```

File: content/modules/redisjson/redisjson-quickstart.md [hunk 2]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+127.0.0.1:12543> JSON.DEL shopping-list
+1
+127.0.0.1:12543> EXISTS shopping-list
+0
 ```
 
-Add another entity to the test document
+## RedisJSON with Python
+
+If you want to use RediSearch within an application, you can use one of these [client libraries](https://redis.io/docs/stack/json/clients/).
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:229]:
```suggestion
If you want to use RedisJSON within an application, you can use one of these [client libraries](https://redis.io/docs/stack/json/clients/).
```

File: content/modules/redisjson/redisjson-quickstart.md [hunk 3]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+        {
+            "name":"socks",
+            "count":4
+        }
+    ]
+]
 ```
 
-Next increment that value by 2 and this returns the updated value.
+Use `JSON.TYPE` to check the JSON type of the key, or an entity inside the key.
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:174]:
```suggestion
Use `JSON.TYPE` to check the JSON type of the key or an entity inside the key.
```

File: content/modules/redisjson/redisjson-quickstart.md [hunk 4]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+print("Creating Shopping list...")
+list_obj = {
+    'list-date': '05/05/2022'
+}
+
+r.json().set('shopping-list:py', '.', list_obj)
+reply = r.json().get('shopping-list:py', '.')
+print(json.dumps(reply, indent=4) + "\n")
+
+# Adding info to the JSON document
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:253]:
```suggestion
# Add info to the JSON document
```

File: content/modules/redisjson/redisjson-quickstart.md [hunk 5]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+    "clothing-store" : {
+        "items": [ { "name": "socks", "count": 2 } ]
+    }
+}
+
+r.json().set('shopping-list:py', '.stores', stores_obj)
+reply = r.json().get('shopping-list:py', '.')
+print(json.dumps(reply, indent=4) + "\n")
+
+# Adding new items to the list
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:271]:
```suggestion
# Add new items to the list
```

File: content/modules/redisjson/redisjson-quickstart.md [hunk 6]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+    "name" : "pears",
+    "count" : 3
+}
+
+r.json().arrappend('shopping-list:py', '.stores.grocery-store.items',
+                    pears_obj)
+reply = r.json().get('shopping-list:py', '.')
+print(json.dumps(reply, indent=4) + "\n")
+
+# Incrementing numerical values
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:283]:
```suggestion
# Increment item counts
```

File: content/modules/redisjson/redisjson-quickstart.md [hunk 7]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+print("Changing item counts...")
+r.json().numincrby('shopping-list:py',
+                    '.stores.clothing-store.items[0].count', 2)
+r.json().numincrby('shopping-list:py',
+                    '.stores.grocery-store.items[1].count', -1)
+reply = r.json().get('shopping-list:py', '.')
+print(json.dumps(reply, indent=4) + "\n")
 
+# Getting all items no matter which heading they're under
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:292]:
```suggestion
# Get all items no matter which heading they're under
```

File: content/modules/redisjson/redisjson-quickstart.md [hunk 8]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+print("Getting all items...")
+reply = r.json().get('shopping-list:py', '$..items')
+print(json.dumps(reply, indent=4) + "\n")
+
+# Deleting portions of the document
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:297]:
```suggestion
# Delete specific parts of the document
```

File: content/modules/redisjson/redisjson-quickstart.md [hunk 9]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+print("Deleting clothing store and pears...")
+r.json().delete('shopping-list:py', '.stores.clothing-store')
+r.json().delete('shopping-list:py', '.stores.grocery-store.items[1]')
+reply = r.json().get('shopping-list:py', '.')
+print(json.dumps(reply, indent=4) + "\n")
+
+# Deleting the key
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:304]:
```suggestion
# Delete the JSON document key
```

File: content/modules/redisjson/redisjson-quickstart.md [hunk 10]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+```
+
+Example output:
 ```sh
-127.0.0.1:12543> JSON.SET foo .test.test3 '"test"'
-OK
-127.0.0.1:12543> JSON.GET foo
-{"foo":"bar","test":{"test2":true,"test3":"test"}}
+$ python3 quick_start.py
+Creating Shopping list...
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:313]:
```suggestion
Creating shopping list...
```

File: content/modules/redisjson/redisjson-quickstart.md [hunk 11]
```diff
@@ -7,98 +7,469 @@ alwaysopen: false
...
+``` python
+import redis
+import json
+
+# Connect to a database
+r = redis.Redis(host="<endpoint>", port="<port>",
+    password="<password>")
+
+# Create a JSON document
+print("Creating Shopping list...")
```
Comment (rrelledge) [inline content/modules/redisjson/redisjson-quickstart.md:244]:
```suggestion
print("Creating shopping list...")
```

**Review by rrelledge — CHANGES_REQUESTED** [review]
Nice updates! This looks good to me. I just left some small suggestions.

File: content/modules/redisjson/redisjson-quickstart.md (hunk 1 above)
Comment (nermiller) [inline content/modules/redisjson/redisjson-quickstart.md:26]:
```suggestion
The [`redis-cli`](https://redis.io/docs/manual/cli/) command-line tool is part of the Redis installation. You can use it to connect to your Redis database and test RedisJSON commands.
```

File: content/modules/redisjson/redisjson-quickstart.md (hunk 2 above)
Comment (nermiller) [inline content/modules/redisjson/redisjson-quickstart.md:229]:
```suggestion
If you want to use RedisJSON within an application, you can use one of the [client libraries](https://redis.io/docs/stack/json/clients/).
```

**Review by nermiller — APPROVED** [review]
LGTM, with just a couple of minor edits. Such nice work, Cameron!

**Review by rrelledge — APPROVED** [review]
Looks good to me.

**Pull request merged**